### ISSUE TITLE
[Jobs] Cheese icon is always showing

### ISSUE THREAD

**Issue by chimera74** [quisquous/cactbot#4032]
## Description

Cheese icon is always shown. No matter the setting.
![image](https://user-images.githubusercontent.com/3358677/149657090-9776ea16-74f1-41c5-a743-9eac69d30cae.png)

## Additional information

Info: Jobs: BrowserConsole: Uncaught TypeError: Cannot read properties of undefined (reading 'foodBuffTimer') (Source: file:///D:/Programs/Advanced%20Combat%20Tracker/Overlay/cactbot/ui/jobs/jobs.bundle.js, Line: 34088)

**Comment by quisquous** [comment]
@MaikoTan @Echoring could one of you look at this? It looks to me like the `return window.setTimeout(this._updateFoodBuff.bind(this), showAfterMs);` line is incorrect because it's not passing any parameters to `updateFoodBuff`.  I wonder if this got broken during some of the jobs refactoring in the past few months.

**Comment by Echoring** [comment]
I cannot reproduce this bug. Cheese icon correctly disappear after I ate food and appear when duration reach setting.
Any other information?

**Comment by quisquous** [comment]
I am just looking at the code.  It seems like the call to `window.setTimeout(this._updateFoodBuff.bind(this), showAfterMs);` will call `_updateFoodBuff` with no arguments (when it requires one).  Does that code look right to you?  I don't understand how it could work.

Do you have an option (e.g. a very large number) set for `HideWellFedAboveSeconds` that prevents this branch from being taken?

**Comment by MaikoTan** [comment]
Yeah, I said the same thing to Echoring yesterday, but she couldn't reproduce this bug before or after the changing. So she asked whether more information.
In this specific line that you mentioned, I think just to add an "o" as the third argument of the `setTimeout` function would solve it. However I don't know if it is related to this bug, either.

**Comment by quisquous** [comment]
I think you can't add `o` as a third argument, because `o` contains some dynamic info, like `inCombat` or `contentType` which may not be the same when the timeout expires.

**Comment by quisquous** [comment]
Repro:
* be in a residential zone on a job that is not max level
* turn off any FC food buffs so that food lasts 30 minutes
* set your cactbot config -> jobs -> Hide cheese icon time to `1740` (i.e. 29 * 60)
* eat food in residential zone
* queue into a dungeon (e.g. Holminster Switch unsynced)
* wait 1 minute
* `[4/9/2022 2:39:13 PM] Info: danabars: BrowserConsole: Uncaught TypeError: Cannot read properties of undefined (reading 'foodBuffTimer')`

In other words you need to eat food in a place where it wouldn't show the cheese icon, then move somewhere where it would when the timer expires.

Additional bugs: based on debug logging, it appears to start a second timeout when zoning, which is incorrect.

**Comment by Echoring** [comment]
> I think you can't add `o` as a third argument, because `o` contains some dynamic info, like `inCombat` or `contentType` which may not be the same when the timeout expires.

This bug appears seems because an undefined `o` pass to the func.
Add `o` like this: `return window.setTimeout(this._updateFoodBuff.bind(this, o), showAfterMs);` indeed solve this bug.

> Additional bugs: based on debug logging, it appears to start a second timeout when zoning, which is incorrect.

This bug appears seems because both zonechange and gaineffect call this func.
When changing zone, buffs which can be taken across zone will "gain" again with the original effect duraion instead of the actual duration remains. I'm still thinking about how to solve this.
